core/native/src/cta/cta_xml.c

```diff
@@ -219,91 +218,74 @@ static CTA_StochObs CTAI_XML_CreateSObs(xmlNode *cur_node) {
-             but the subsobs subtree and the timeoffset  */
-          // make a vector of two handles: the subtree and the offset value
-          retval = CTA_Vector_Create(CTA_DEFAULT_VECTOR, 2, CTA_HANDLE, CTA_NULL, &hvec1);
-          // fill first element
-          retval = CTA_Vector_SetVal(hvec1,1,&hnew_sub,CTA_HANDLE);
-          //fill second element
-          retval = CTA_Vector_Create(CTA_DEFAULT_VECTOR, 1, CTA_DOUBLE, CTA_NULL, &htimeoffset);
-          if (timeoffset){
-             timeoffsetval = atof((char *) timeoffset);
-          } else {
-             timeoffsetval = 0.0;
-          }
-          retval = CTA_Vector_SetVal(htimeoffset,1,&timeoffsetval,CTA_DOUBLE);
-          retval = CTA_Vector_SetVal(hvec1,2,&htimeoffset,CTA_HANDLE);
-
-
-          retval = CTA_SObs_Create(sobsclass, hvec1, &hnew);
-
-        } else {
-          printf("ERROR: neither database given nor combined observer\n");
-          exit(-1);
-        }
-
+   if (database) {
+
+      CTA_String_Set(hdb, (char *) database);
+
+      if (timeoffset){
+         timeoffsetval = atof((char *) timeoffset);
+         // make a vector of two handles: the subtree and the offset value
+         retval = CTA_Vector_Create(CTA_DEFAULT_VECTOR, 2, CTA_HANDLE, CTA_NULL, &hvec1);
+         // fill first element (name of database)
+         retval = CTA_Vector_SetVal(hvec1,1,&hdb,CTA_HANDLE);
+         // fill second element
+         retval = CTA_Vector_Create(CTA_DEFAULT_VECTOR, 1, CTA_DOUBLE, CTA_NULL, &htimeoffset);
+         retval = CTA_Vector_SetVal(htimeoffset,1,&timeoffsetval,CTA_DOUBLE);
+         retval = CTA_Vector_SetVal(hvec1,2,&htimeoffset,CTA_HANDLE);
+         // create sobs
+         retval = CTA_SObs_Create(sobsclass, hvec1, &hnew);
+      } else {
+         // 'old' situation: userdata is name of database
+         retval = CTA_SObs_Create(sobsclass, hdb, &hnew);
       }
-
-
-      CTA_String_Free(&hdb);
-      if (retval == CTA_OK) {
-         /* Perform initial selection */
-         if (selection) {
-            CTA_String_Create(&hsel);
-            CTA_String_Set(hsel, (char *) selection);
-            retval=CTA_SObs_CreateSel(hnew, hsel, &hnew2);
-            CTA_String_Free(&hsel);
-            CTA_SObs_Free(&hnew);
-            if (retval != CTA_OK){
-               return CTA_NULL;
-            }
-            hnew=hnew2;
-         }
-
-         /* Set name (if any) */
-         if (name) {
-            CTAI_Handle_SetName(hnew, (char *) name);
-           // xmlFree(name);
+   }
+   else { //combined CTA_OBS
+      if (is_combined == 1) {
+         /* create the combined SObs  with for the 2nd argument not a database
+            but the subsobs subtree and the timeoffset  */
+         // make a vector of two handles: the subtree and the offset value
+         retval = CTA_Vector_Create(CTA_DEFAULT_VECTOR, 2, CTA_HANDLE, CTA_NULL, &hvec1);
+         // fill first element
+         retval = CTA_Vector_SetVal(hvec1,1,&hnew_sub,CTA_HANDLE);

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'retval' is reassigned a value before the old one has been used.](https://app.codacy.com/gh/OpenDA-Association/OpenDA/pullRequest?prid=5593693)